app/controllers/subscriptions_management_controller.rb

```diff
@@ -0,0 +1,149 @@
+class SubscriptionsManagementController < ApplicationController
+  before_action :set_cache_control_header
+  before_action :check_if_access_is_enabled
+  before_action :get_subscriptions
+
+  MISSING_EMAIL_ERROR = "Please enter your email address.".freeze
+  INVALID_EMAIL_ERROR = "This doesn’t look like a valid email address – check you’ve entered it correctly.".freeze
+
+  def index; end
+
+  def update_frequency
+    @back_url = list_subscriptions_path
+    @subscription_id = subscription_id
+    @current_frequency = @subscriptions[@subscription_id]['frequency']
+  end
+
+  def change_frequency
+    # TODO: Enable this
+    # email_alert_api.change_subscription(
+    #   id: subscription_id,
+    #   frequency: new_frequency
+    # )
+    session[:subscriptions][subscription_id]['frequency'] = new_frequency
+
+    #clear_subscriptions!
+
+    redirect_to list_subscriptions_path
+  end
+
+  def update_address
+    @back_url = list_subscriptions_path
+    @address = @subscriber['address']
+  end
+
+  def change_address
```

Comment:
When we update an email address, is the current plan to just update with no confirmation or are we going to set up a confirmation link to both old and new email addresses before we action it? If we go the paranoid route then changing an address should probably invalidate the session and tell them to do the confirmation dance....

Reply:
I was going down the route of just changing it...